sdk/servicebus/service-bus/src/serviceBusMessage.ts

```diff
@@ -817,9 +817,19 @@ export class ServiceBusMessage implements ReceivedMessage {
       this.messageId
     );
     if (this._context.requestResponseLockedMessages.has(this.lockToken!)) {
+      let receiverName;
+      if (this.sessionId !== undefined) {
+        receiverName = this._context.messageSessions[this.sessionId].name;
```

Comment:
It is not guaranteed that `this._context.messageSessions[this.sessionId]` would exist. The `messageSession` for the session could have been closed in which case `this_context.messageSessions` map will not have an entry for `this.sessionId`.

Also, we are repeating this logic of finding the associated link name in 4 places. Can we create a private helper method, say `getAssociatedLinkName()` instead?

Reply:
If we make such a helper method, then we can use it in all other places, where we do similar if/else checks to get the associated link name. Here is my proposal, let me know what you think

- Export 2 helpers from the `clientEntityContext.ts` (or other file as you see fit)
     - `getAssociatedReceiverLinkName(clientContext: ClientEntityContext, sessionId?: string)`
     - `getAssociatedSenderLinkName(clientContext: ClientEntityContext)`
- All the functions in `ManagementClient` class that need the `associatedLinkName` can call one of the above instead of expecting the caller to pass in the value.

Thoughts?

Reply:
About the sessionId not being present when it's expired, in such cases there is no associated link name to pass. Hence, the implementations holds good.

About refactoring to use these 2 additions on the interface, I think it would need a design discussion and further efforts. For this PR, we can unblock ourselves with current changes and take it up later?

Reply:
> About the sessionId not being present when it's expired, in such cases there is no associated link name to pass. Hence, the implementations holds good.

I was referring to the error we would get when trying to get the `name` i.e
for the session expired scenario, in the below code, we will get "`name` doesnt exist on `undefined`" error since `this._context.messageSessions[this.sessionId]` will return `undefined`

```javascript
if (this.sessionId !== undefined) {
      receiverName = this._context.messageSessions[this.sessionId].name;
    } 
```

> About refactoring to use these 2 additions on the interface, I think it would need a design discussion

The helper functions need not be on the interface of ClientEntityContext. The way I have suggested above, they can be stand-alone helper functions and not inside any interface or class. 

> For this PR, we can unblock ourselves with current changes and take it up later?

At present we have the `_getAssociatedReceiverName()` as part of this PR to reduce code duplication in 4 places. With slight modification (take `context` and `sessionId` as input parameters instead of using from `this`), the same helper can be used from 7 other places (3 in `Receiver`, 2 in `QueueClient` and 2 in `SubscriptionClient`). We should atleast pick this up for this PR.

Updating the functions in the `ManagementClient` class to use this helper (and another for sender) instead of taking `associatedLinkName` can be taken up later.

Reply:
Oh okay, updating to include the undefined check on `messageSessions`.

Adding the helper function to utils.ts and updating the receiver related references
(Adding it to ClientEntityContext seemed confusing/difficult at first since this was bookkeeping a lot of things.)